packages/design-system/src/components/Tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,337 @@
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+      document.removeEventListener('keydown', handleEscapeKey);
+    };
+  }, [handleClickOutside, handleEscapeKey]);
+
+  useEffect(() => {
+    if (active) {
+      props.onOpen && props.onOpen();
+    } else {
+      props.onClose && props.onClose();
+    }
+  }, [active]);
+
+  useLayoutEffect(() => {
+    if (popper.current) {
+      popper.current.setOptions(props);
+      popper.current.forceUpdate();
+    }
+  });
+
+  const renderTrigger = (props: ITooltipProps): React.ReactElement => {
+    const {
+      activeClassName,
+      ariaLabel,
+      children,
+      className,
+      component,
+      dialog,
+      offset,
+      onClose,
+      onOpen,
+      inversed,
+      interactiveBorder,
+      placement,
+      maxWidth,
+      title,
+      transitionDuration,
+      zIndex,
+      ...others
+    } = props;
+
+    const TriggerComponent = component;
+    const triggerClasses = classNames('ds-base', 'ds-c-tooltip__trigger', className, {
+      [activeClassName]: active,
+      'ds-c-tooltip__trigger--inverse': inversed,
+    });
+    const linkTriggerOverrides = {
+      role: TriggerComponent === 'a' ? 'button' : undefined,
+      tabIndex: TriggerComponent === 'a' ? 0 : undefined,
+    };
+    const eventHandlers = dialog
+      ? {
+          onTouchStart: () => handleTouch(),
+          onClick: () => {
+            if (!isMobile) {
+              setActive(!active);
+            }
+          },
+        }
+      : {
+          onTouchStart: () => handleTouch(),
+          onClick: () => {
+            if (!isMobile) {
+              setActive(!active);
+            }
+          },
+          onFocus: () => setActive(true),
+          onBlur: (event) => handleBlur(event),
+        };
+
+    return (
+      <TriggerComponent
+        type={TriggerComponent === 'button' ? 'button' : undefined}
+        aria-label={ariaLabel || undefined}
+        aria-describedby={dialog ? undefined : id.current}
+        className={triggerClasses}
+        ref={setTriggerElement}
+        {...others}
+        {...linkTriggerOverrides}
+        {...eventHandlers}
+      >
+        {children}
+      </TriggerComponent>
+    );
+  };
+
+  const renderContent = (props: ITooltipProps): React.ReactElement => {
+    const {
+      dialog,
+      inversed,
+      interactiveBorder,
+      placement,
+      maxWidth,
+      title,
+      transitionDuration,
+      zIndex,
+    } = props;
+
+    const tooltipStyle = { maxWidth, zIndex };
+    const interactiveBorderStyle: React.CSSProperties = {
+      left: `-${interactiveBorder}px`,
+      top: `-${interactiveBorder}px`,
+      border: `${interactiveBorder}px solid transparent`,
+      zIndex: -999, // ensures interactive border doesnt cover tooltip content
+    };
+    const eventHandlers = dialog ? {} : { onBlur: (event) => handleBlur(event) };
+
+    const tooltipContent = () => (
```

Comment:
This could be just a value instead of a function.

Reply:
The parent function `renderContent` I kind of feel the same way about, but there's probably enough that has changed from the original already.

Reply:
Agreed, I had actually flattened the whole thing out at one point, since this is re-used I figured I would leave it instead of wrapping the focus element in conditionals..